### ISSUE TITLE
0.9 -> 0.10 migration errors

### ISSUE THREAD

**Issue by djeedai** [bevyengine/bevy-website#603]
(reopened from bevyengine/bevy#8033)

Here's a list of migration errors with no corresponding items in the migration guide:

- The stageless migration really doesn't explain the basics of how to migrate from _e.g._ `add_system_to_stage()` to ... (?) There's not even a single example.
- `ReportExecutionOrderAmbiguities` (likely: `bevy::ecs::schedule::ScheduleBuildSettings::ambiguity_detection`)
- `ModifiesWindows` (?)
- `StageLabel` (?)
- `MonitorSelection` moved to inner of `WindowPosition`, `Window::monitor` field deleted
- `App::add_system_set()` removed, how to replace?
- `ResMut<State<X>>` doesn't build anymore; this needs the new `States` trait which is not mentioned in the migration guide
- `State::current()` (likely `state.0`). (`NextState` is mentioned but it's not super clear either)

(TODO : didn't finish my update)

**Comment by Weibye** [comment]
Additionally: How exactly should one setup systems that previously ran with a fixed timestep?

**Comment by denormative** [comment]
It took half a day to upgrade the relatively simple roguelike I was working on and there's no way I could have done it without finding these notes online: https://zenn.dev/hyoi/scraps/5ea4bc5c9f6436 And I've still got a weird bug where the action N only triggers after I've triggered the N+1's action, but I'm guessing that's likely because of needing use NextState instead of just pushing the new state on the stack like I used to.

Anyway, I get how this new stateless system is more flexible and such, but I think the upgrade documentation was way to minimal to be of use to people not following the conversions in the PRs/issues.

**Comment by djeedai** [comment]
> Anyway, I get how this new stateless system is more flexible and such, but I think the upgrade documentation was way to minimal to be of use to people not following the conversions in the PRs/issues.

Yes I agree. I _feel_ reading some descriptions the potential stageless has, and I'm pretty sure it's much better than stages, but in practice being the number one feature of this release, a complete rework of a fundamental system, and not having a single migration example in the migration guide is quite frustrating.

@alice-i-cecile I hope we can fix that guide quickly before too many people struggle. I can try to help but given I genuinely don't understand yet how to fix most of the items listed I won't be able to make a PR for now. Is anyone with ECS insight able to jump on this please?

**Comment by alice-i-cecile** [comment]
I'll do my best to tackle this in the next couple of days.

**Comment by denormative** [comment]
Update seems good. I'm redoing my update from scratch following the guide since I botched it last time, I am noticing there's a few things missing though like the old `SystemSet::on_inactive_update`. I dug into the RFC and it mentions these, but they're not called out as being removed anywhere else, or the fact there's apparently no equivalent for them.

For `SystemSet::on_inactive_update` I assume it's just a case of adding the systems to all the other states as well, but it might be worth copying the relevant section from the RFC to the migration guide as well since google was particularly unhelpful finding me the exact details.

**Comment by alice-i-cecile** [comment]
In that case I would make a custom run condition, which checks if the state resource is not equal to the specified state :) Much more flexible and easier to ensure it's correct!

**Comment by djeedai** [comment]
I found https://github.com/bevyengine/bevy/issues/7258 back from January to add to the list of missing things. I'm a bit surprised this was cut from release given it clearly highlights that users _will_ get confused. In fact until I read the issue I had no idea this could ever be an error. And I think this shows I still don't understand `SystemSet`.
